design-documents/graph-ql/coverage/gift-registry.graphqls

```diff
@@ -0,0 +1,296 @@
+    email: String
+    dynamic_attributes: [GiftRegistryDynamicAttributeInput] @doc(description: "As a result of the update, only the values of provided attributes will be affected. If the attribute is missing in the request, its value will not be changed")
+}
+
+type CreateGiftRegistryRegistrantInput {
+    first_name: String!
+    last_name: String!
+    email: String!
+    dynamic_attributes: [GiftRegistryDynamicAttributeInput]
+}
+
+type GiftRegistryDynamicAttributeInput {
+    code: ID!
+    value: String!
+}
+
+type CreateGiftRegistryOutput {
+    gift_registry: GiftRegistry
+}
+
+type UpdateGiftRegistryOutput {
+    gift_registry: GiftRegistry
+}
+
+type AddGiftRegistryItemsOutput {
+    gift_registry: GiftRegistry
+}
+
+type RemoveGiftRegistryItemsOutput {
+    gift_registry: GiftRegistry
+}
+
+type UpdateGiftRegistryItemsOutput {
+    gift_registry: GiftRegistry
+}
+
+type GiftRegistryType {
+    id: ID!
+    label: String!
+    dynamic_attributes_metadata: [GiftRegistryDynamicAttributeMetadataInterface]
+}
+
+interface GiftRegistryDynamicAttributeMetadataInterface {
+    code: ID!
+    input_type: String!
+    attribute_group: String!
+    label: String!
+    is_required: Boolean!
+    sort_order: Int
+}
+
+type GiftRegistryTextAttributeMetadata implements GiftRegistryDynamicAttributeMetadataInterface {
+}
+
+type GiftRegistrySelectAttributeMetadata implements GiftRegistryDynamicAttributeMetadataInterface, GiftRegistrySelectAttributeMetadataInterface {
+}
+
+type GiftRegistryDateAttributeMetadata implements GiftRegistryDynamicAttributeMetadataInterface {
+    format: GiftRegistryDateAttributeFormat!
+}
+
+enum GiftRegistryDateAttributeFormat {
+    SHORT
+    MEDIUM
+    LONG
+    FULL
+}
+
+type GiftRegistryCountryAttributeMetadata implements GiftRegistryDynamicAttributeMetadataInterface {
+    show_region: Boolean!
+}
+
+type GiftRegistryEventCountryAttributeMetadata implements GiftRegistryDynamicAttributeMetadataInterface, GiftRegistrySearcheableAttributeMetadataInterface {
+    show_region: Boolean!
+}
+
+type GiftRegistryEventDateAttributeMetadata implements GiftRegistryDynamicAttributeMetadataInterface, GiftRegistrySearcheableAttributeMetadataInterface {
+    format: GiftRegistryDateAttributeFormat!
+}
+
+type GiftRegistryEventLocationAttributeMetadata implements GiftRegistryDynamicAttributeMetadataInterface, GiftRegistrySearcheableAttributeMetadataInterface {
+}
+
+type GiftRegistryRoleAttributeMetadata implements GiftRegistryDynamicAttributeMetadataInterface, GiftRegistrySearcheableAttributeMetadataInterface, GiftRegistrySelectAttributeMetadataInterface {
+}
+
+interface GiftRegistrySearcheableAttributeMetadataInterface {
+    is_searcheable: Boolean!
+}
+
+interface GiftRegistrySelectAttributeMetadataInterface {
+    options: [GiftRegistrySelectAttributeOptionMetadata]
+}
+
+type GiftRegistrySelectAttributeOptionMetadata {
+    code: ID!
+    label: String!
+    is_default: Boolean
+}
+
+type GiftRegistry {
+    id: ID!
```

Comment:
Is there a possibility to share a gift registry through the shareable token or URL?

Reply:
Yes, see scenarios I'm aware of in another document (same PR). Schema is still in progress, will add support for this flow soon.